frontend/src/features/properties/map/research/update/models.ts

```diff
@@ -0,0 +1,120 @@
+  public id?: number;
+  public name?: string;
+  public roadName?: string;
+  public roadAlias?: string;
+  public rfileNumber?: string;
+  public statusTypeCode?: string;
+  public statusTypeDescription?: string;
+  //public researchProperties?: Api_ResearchFileProperty[];
+  public requestDate?: string;
+  public requestDescription?: string;
+  public requestSourceDescription?: string;
+  public researchResult?: string;
+  public researchCompletionDate?: string;
+  public isExpropriation?: string;
+  public expropriationNotes?: string;
+  public requestSourceTypeCode?: string;
+  public requestSourceTypeDescription?: string;
+  public requestor?: IContactSearchResult;
+  public researchFilePurposes?: ResearchFilePurposeFormModel[];
+  public rowVersion?: number;
+
+  public static fromApi(base: Api_ResearchFile): UpdateResearchFormModel {
+    var model = new UpdateResearchFormModel();
+    model.id = base.id;
+    model.name = base.name;
+    model.roadName = base.roadName;
+    model.roadAlias = base.roadAlias;
+    model.rfileNumber = base.rfileNumber;
+    model.statusTypeCode = base.statusTypeCode?.id;
+    model.statusTypeDescription = base.statusTypeCode?.description;
+    //model.researchProperties = base.researchProperties;
+    model.requestDate = base.requestDate;
+    model.requestDescription = base.requestDescription;
+    model.requestSourceDescription = base.requestSourceDescription;
+    model.researchResult = base.researchResult;
+    model.researchCompletionDate = base.researchCompletionDate;
+    model.isExpropriation = base.isExpropriation ? 'true' : 'false';
+    model.expropriationNotes = base.expropriationNotes;
+    model.requestSourceTypeCode = base.requestSourceType?.id;
+    model.requestSourceTypeDescription = base.requestSourceType?.description;
+
+    console.log(base.requestorPerson);
+    console.log(base.requestorOrganization);
+
+    if (base.requestorPerson !== undefined) {
+      model.requestor = fromApiPerson(base.requestorPerson);
+    } else if (base.requestorOrganization !== undefined) {
+      model.requestor = fromApiOrganization(base.requestorOrganization);
+    }
+    model.researchFilePurposes = base.researchFilePurposes?.map(x =>
+      ResearchFilePurposeFormModel.fromApi(x),
+    );
+    model.rowVersion = base.rowVersion;
+    return model;
+  }
+
+  public toApi(): Api_ResearchFile {
+    console.log(this.id);
+    return {
+      id: this.id,
+      name: this.name,
+      roadName: this.roadName,
+      roadAlias: this.roadAlias,
+      rfileNumber: this.rfileNumber,
+      statusTypeCode: { id: this.statusTypeCode },
+      //researchProperties: this.researchProperties,
+      requestDate: this.requestDate,
+      requestDescription: this.requestDescription,
+      requestSourceDescription: this.requestSourceDescription,
+      researchResult: this.researchResult,
+      researchCompletionDate: this.researchCompletionDate,
+      isExpropriation:
+        this.isExpropriation === undefined
```

Comment:
this seems like a good candidate for the use of a utility function. we may already have one for this.